### ISSUE TITLE
Vaadin Spring boot app in the npm mode (Vaadin 14) without any plugin ?

### ISSUE THREAD

**Issue by vlipovetskii** [vaadin/flow#6277]
https://vaadin.com/tutorials/migrate-from-bower-to-npm-for-vaadin-14+

Aziz Rehman "Hi, we have here a Gradle-Multi-Project, it seems to be working currently in the bower compatibility mode with latest vaadin-14.X.rc. However, I am not sure how do I migrate it to npm. Any Tips would be really appreciated. Apparently, there is no vaadin-gradle plugin that takes care of genrating npm files as mention in :"2.1 Add front-end plugin"

It would be great for if you could mention what files and where they should be existed so that the npm works. I am not using any vaadin-gradle plugin uptill now and do not require do so."

Like Aziz Rehman, I build Vaadin projects with gradle and without any specific vaadin-gradle plugin. How to build Vaadin Spring boot app in the npm mode without any plugin ?

**Comment by caalador** [comment]
For a Spring boor gradle application it should be enough to add into `application.properties` the property `vaadin.compatibilityMode = false`. This should make the project start in npm mode and generate all the required files as long as node and npm is installed.

**Comment by vlipovetskii** [comment]
@caalador  Thank you ! It works, if vaadin.productionMode=false

After I set vaadin.productionMode=true, another error
java.io.IOException: The stats file from webpack (stats.json) was not found.

**Comment by vlipovetskii** [comment]
I have created an app using https://vaadin.com/start/latest
And it is the same problem when vaadin.productionMode=true

**Comment by vlipovetskii** [comment]
BTW, I have migrated to 14.0.1 - still the problem

**Comment by vlipovetskii** [comment]
The similar issue: https://vaadin.com/forum/thread/17763152/vaadin-14-npm-mode-production-profile

**Comment by caalador** [comment]
For a production mode build you would need to run webpack to generate the client bundle.

Simplest would be to after getting all the files from starting in development mode run `node node_modules\webpack\bin\webpack.js` in the project root which will create the project bundles and stats.json file.

For an automated gradle production mode build from scratch I can't really help and will have to wait until the gradle-vaadin-plugin gets the support added.

**Comment by vlipovetskii** [comment]
I have tried. 

stats.json has been generated in config folder where application.properties is located.
I copied stats.json to the \src\main\resources\META-INF\VAADIN\config\
App starts without error: Cannot get the 'stats.json' from the classpath 'META-INF/VAADIN/config/stats.json', but displays text chunk  instead of correct layout.
Pathes inside stats.json look like ""name": "build/vaadin-bundle-6889b06816d4db802698.cache.js"
It seems, vaadin can't find resources from stats.json.

I tried to put stats.json into project root. It didn't help.

What shoud I fix ?

**Comment by caalador** [comment]
What resource(s) couldn't be found?

**Comment by vlipovetskii** [comment]
E.g.
<script type="text/javascript" src="./VAADIN/build/webcomponentsjs/webcomponents-loader.js">

webpack.js created webcomponentsjs/* and other resources in the <project root>/build

I have created VAADIN/build in the <project root> and copied webpack generated stuff there - still the same issue.

**Comment by caalador** [comment]
To get it running you should copy `{ROOT}/config` to `src/main/resources/META-INF/VAADIN/` 
And from the `{ROOT}/build` copy `webcomponentsjs/**` and `vaadin-bundle*.js` to `src/main/resources/META-INF/VAADIN/build`  

This way it should be able to run in production mode

**Comment by vlipovetskii** [comment]
Thank you very much ! It works now !

Is there a command line parameter for webpack.js to generate stats.json and other bundles in the correct folders ? If no, I will create my own bat to automate copying.

**Comment by caalador** [comment]
The `webpack.generated.js` configuration file has the constant `const mavenOutputFolderForFlowBundledFiles = require('path').resolve(__dirname, '');` this determines where the files are created. I'm not certain if you can do a override in the `webpack.config.js` with:

`flowDefaults.output.path = require('path').resolve(__dirname, 'src/main/resources/META-INF/VAADIN/');`

The problem with possibly editing the generated file is that any later run in dev mode will override the file where as the config file will be untouched.

**Comment by vlipovetskii** [comment]
I see. It's simpler to create a separate script to run webpack.js and copy generated files to the correct folders. 
Thank you for your help !

**Comment by caalador** [comment]
Closing issue as the proper fix will later come into the gradle plugin.

**Comment by adispennette** [comment]
Are there any details that can be provided for the plugin?  I have a boot existing app that I am trying to integrate vaadin into and I get 404's when attempting to locate VAADIN/build/vaadin-bundle-3dbea72c8cabf4cbd7ae.cache.js  

I have tried adding everything to META-INF/VAADIN/build but it is still not found
I have also tried adding it under BOOT-INF/classes/META-INF/VAADIN/build and nothing
I did a quick google of the gradle plugin you mentioned but it doesn't seem like its ready to do what is expected in this issue. (or at least the documentation seems to be lacking in explaining anything)

I know this item is closed but I believe it is still an issue and needs some more explanation and documentation for a workaround.

thanks
